desktop/cmp/dash/DashViewModel.js

```diff
@@ -65,4 +66,30 @@ export class DashViewModel {
 
         this.refreshContextModel = new DashRefreshContextModel(this);
     }
+
+    setTitle(title) {
+        this.setViewStateKey('title', title);
+    }
+
+    setIcon(icon) {
+        this.setViewStateKey('icon', icon);
+    }
+
+    /**
+     * Modify a single key on this model's viewState
+     */
+    setViewStateKey(key, value) {
+        this.setViewState({...this.viewState, [key]: value});
+    }
+
+    /**
+     * Overwrite viewState, preserving any existing title or icon
+     */
+    @action
+    setViewState(viewState) {
+        const {title, icon} = this.viewState ?? {},
+            newState = {title, icon, ...viewState};
+        this.viewState = pickBy(newState, v => !isNil(v));
+    }
```

Comment:
my only concern here tom is:  what if I actually want to store null as a key value in my own custom state -- I should be able to do that

Reply:
did we consider storing title and icon *next* to viewState?  Then if we had a widget that has no viewState (but the user set the title and icon) we would have no viewState.

Reply:
Good idea - I agree that at this point, now we are passing the viewModel to the view, it makes sense to separate title and icon into their own distinct things. Think it will make it cleaner all round. Will have a go at that.